packages/toolkit/src/query/core/buildMiddleware/devMiddleware.ts

```diff
@@ -1,47 +1,34 @@
-import type { SubMiddlewareBuilder } from './types'
+import type { InternalHandlerBuilder } from './types'
 
-export const build: SubMiddlewareBuilder = ({
+export const buildDevCheckHandler: InternalHandlerBuilder = ({
   api,
   context: { apiUid },
   reducerPath,
 }) => {
-  return (mwApi) => {
-    let initialized = false
-    return (next) => (action) => {
-      if (!initialized) {
-        initialized = true
-        // dispatch before any other action
-        mwApi.dispatch(api.internalActions.middlewareRegistered(apiUid))
-      }
-
-      const result = next(action)
-
```

Comment:
If all of this went into the initialization, should maybe the rest of it also go there? Like this, the logic is pretty split up between the files.

Reply:
I could, yeah. Conceptually it felt like this file was split between "one-time init" and "dev check warning", so I left the dev warning in here.